File: Api/Models/Category.cs
namespace Api.Models
{
    public class Category
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public Interval Interval { get; set; } = Interval.Once;
        public bool IsDefault { get; set; } = false;
        // Verknüpfung zum User (Jeder User hat eigene Kategorien)
        public int UserId { get; set; }
        public User User { get; set; } = null!;
        public List<Keyword> Keywords { get; set; } = [];
    }
}

File: Web/src/types/category.ts
import { type Keyword, mapKeyword } from "./keyword";


export type Category = {
    id: number;
    name: string;
    keywords: Keyword[];
    interval: string;
    isDefault: boolean;
}

export function mapCategory(data: any): Category {
    return {
        id: data.id,
        name: data.name,
        keywords: data.keywords.map((k: any) => mapKeyword(k)),
        interval: data.interval,
        isDefault: data.isDefault
    };
}